src/particle_restart.cpp

```diff
@@ -95,13 +95,11 @@ void run_particle_restart()
   int64_t particle_seed;
   switch (previous_run_mode) {
   case RunMode::EIGENVALUE:
+  case RunMode::FIXED_SOURCE:
```

Comment:
I'm glad this is the approach you settled on as it simplifies the code (to me) and is agnostic to the run mode.